### PR TITLE
Add creation time in tag list page

### LINKED ISSUE #21699: Show dates for tags
### Feature Description

When looking at tags, only their name and commit sha are shown, but not their date.

Showing the dates for tags would be super useful to understand their age, or history of a project at a glance.

### Screenshots

image

image

### PR DESCRIPTION
**Pull request by balki**
Fixes #21699

### CODE REVIEW

File: templates/repo/release/list.tmpl
Comment (sillyguodong) [inline templates/repo/release/list.tmpl]: `gr-mr-3` => `gt-mr-3`

Comment (balki) [reply templates/repo/release/list.tmpl]: oops! fixed. Thanks!

**Review by silverwind — APPROVED** [review]

**Review by sillyguodong — APPROVED** [review]

File: templates/repo/release/list.tmpl
Comment (wxiaoguang) [inline templates/repo/release/list.tmpl]: Wait.

1. What does "time gt-mono muted" mean here? I guess many (all) of them are misused.
2. Take a look at #21440
    * ~~The time should follow the best practice at the moment.~~
    * Hmm ... the `formatAllTimeElements` doesn't support relative time, so this comment could be ignored at the moment, leave the relative time support to future. ref: [RelativeTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat)

Comment (silverwind) [reply templates/repo/release/list.tmpl]: - `gt-mono` seems unnessary, it's only meant for the commit hash to render in monospace font
- `muted` is needed for the link color

Comment (wxiaoguang) [reply templates/repo/release/list.tmpl]: But it isn't a link. It is a `span`.

Comment (silverwind) [inline templates/repo/release/list.tmpl]:
```suggestion
											<span class="time gt-mr-3">{{svg "octicon-clock" 16 "gt-mr-2"}}{{TimeSinceUnix .CreatedUnix $.locale}}</span>
```

**Review by silverwind — CHANGES_REQUESTED** [review]
comment above

File: templates/repo/release/list.tmpl
Comment (wxiaoguang) [reply templates/repo/release/list.tmpl]: Copy above:

* But it isn't a link. It is a `span`. So I guess the `muted` is also unnecessary?
* What does the `time` class do? I guess there is no style or JS for it?

Comment (silverwind) [reply templates/repo/release/list.tmpl]: Right, `muted` is completely unnecessary then.

I think class should probably be `time-since` as that should cause JS to add the tooltip to it. Also add other attributes:

````html
<span class="time-since" data-tooltip-content="%s" data-tooltip-interactive="true">%s</span>
````

Comment (wxiaoguang) [reply templates/repo/release/list.tmpl]: Nope, there is no such JS. In #21440, we all agreed to use `<time>` to make everything clear. There is no code for old `time-since`

<details>

image

</details>

Comment (silverwind) [reply templates/repo/release/list.tmpl]: BTW there are more similar cases in templates which should be adapted to above syntax. For example `templates/repo/release/list.tmpl`:

````
<span class="time">{{TimeSinceUnix .CreatedUnix $.locale}}</span>
````

Comment (silverwind) [reply templates/repo/release/list.tmpl]: The class name does not matter but the `data-tooltip-content` and `data-tooltip-interactive` should be added as per https://github.com/go-gitea/gitea/pull/23526. But I do agree we should probably unify the class name as well.

Comment (wxiaoguang) [reply templates/repo/release/list.tmpl]: That's different. This page is not `.repository.list`.

Actually, most `TimeSincUnix` in templates doesn't have/need special CSS classes.

<details>

image

</details>

Comment (wxiaoguang) [reply templates/repo/release/list.tmpl]: I would suggest to remove all of these unnecessary/unused classes ( "time gt-mono muted" ), otherwise they would cause technical debt or potential conflicts.

In the future, we can use some relative-time Custom Element to replace all these `TimeSinceUnix`

Comment (balki) [reply templates/repo/release/list.tmpl]:
```suggestion
											<span class="gt-mr-3 gt-mono">{{svg "octicon-clock" 16 "gt-mr-2"}}{{TimeSinceUnix .CreatedUnix $.locale}}</span>
```
Does this look good? I think `gt-mono and `gt-mr3` are still needed to keep it consistent to rest of the line. Is there any way to run a demo from master and see how it looks? My installation is still in 1.18.5 where it looks like below (without the gt-prefix). 
image

Comment (silverwind) [reply templates/repo/release/list.tmpl]: Hmm, no the current view is this:

image

I think if you just do single class `gt-mr-3`, it should be fine. We don't want monospace font there as only the commit hash is (correctly) using it.

Comment (silverwind) [reply templates/repo/release/list.tmpl]: > Actually, most TimeSincUnix in templates doesn't have/need special CSS classes.

I think for preparation of future JS-derived times (whih respect user timezone, unlike the current server solution), it'll be good to have this class name as a preparation.

Anyways, I will try this PR later and will probably do the necessary class name changes, @balki no need to touch it right now.

Comment (silverwind) [reply templates/repo/release/list.tmpl]: Fixed in my latest commit.

**Comment by silverwind** [comment]
Pushed a fix that removed the unnecessary classes. What eluded me is that `TimeSinceUnix` already renders a `<span>`, so not time class needed, but wrapper span is still needed for margin.

image

This is how it looks now:

image

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Adding backport label as it's a nice litte feature with no risk attached.

**Comment by balki** [comment]
@silverwind  I don't have merge permissions. Please merge if everything looks good. Let me know if I have to rebase/merge from master.

**Comment by silverwind** [comment]
Mergers will merge it when they get around to it.

**Pull request merged**
